database/mysql/mysql.go

```diff
@@ -327,8 +340,15 @@ func (m *Mysql) Run(migration io.Reader) error {
 		return err
 	}
 
+	ctx := context.Background()
+	if m.config.StatementTimeout != 0 {
```

Comment:
I didn't catch this in the postgres PR for this but is there a valid case for negative timeouts? e.g. would anyone want all statements to timeout immediately?

Reply:
Perhaps someone would want to in some testing situation, where they exercise the behaviour of `migrate` in some sort of integration test and want to ensure it times out for a long running migration. For this case you wouldn't need a negative timeout though, an `x-statement-timeout` of say 500ms and a migration containing `DO SLEEP(5);` would suffice in that case.

I can't think of any other situations, can you?

Reply:
> some sort of integration test and want to ensure it times out for a long running migration

Sounds like a valid case to me. Let's continue to allow negative statement timeouts